index.js

```diff
@@ -1014,6 +1024,112 @@ function getChoppedStreamHandler (client, onerror) {
   }
 }
 
+/**
+ * Some behaviors in the APM depend on the APM Server version. These are
+ * exposed as `Client#supports...` boolean methods.
+ *
+ * These `Client#supports...` method names intentionally match those from the Java agent:
+ * https://github.com/elastic/apm-agent-java/blob/master/apm-agent-core/src/main/java/co/elastic/apm/agent/report/ApmServerClient.java#L322-L349
+ */
+Client.prototype.supportsKeepingUnsampledTransaction = function () {
+  // Default to assuming we are using a pre-8.0 APM Server if we haven't
+  // yet fetched the version. There is no harm in sending unsampled transactions
+  // to APM Server >=v8.0.
+  if (!this._apmServerVersion) {
+    return true
+  } else {
+    return this._apmServerVersion.major < 8
+  }
+}
+
+/**
+ * Fetch the APM Server version and set `this._apmServerVersion`.
+ * https://www.elastic.co/guide/en/apm/server/current/server-info.html
+ *
+ * If fetching/parsing fails then the APM server version will be set to `null`
+ * to indicate "unknown version".
+ */
+Client.prototype._fetchApmServerVersion = function () {
+  const self = this
+
+  const setVerUnknownAndNotify = (errmsg) => {
```

Comment:
nit/question: With the `() => {}` style function the use of `self` shouldn't be necessary.  Any particular reason/mental-algorithm for making this an arrow function?

Reply:
This old dog is slowing attempting to master new tricks? ;)
I can drop the `self`ing.